sys/include/net/ng_netapi.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_netapi Generic network module interface
+ * @ingroup     net
+ * @brief       Generic interface for IPC communication between network modules
+ *
+ * @details     The idea of this interface is that it provides every network
+ *              module with a basic set of commands to communicate with its
+ *              neighboring modules. In this model every module runs in its own
+ *              thread and communication is done using netapi.
+ *
+ *                     +-----+  +-----+   +-----+
+ *                     | App |  | App |   | App |
+ *                     +-----+  +-----+   +-----+
+ *                        ^        ^         ^
+ *                      netapi   netapi    netapi
+ *                        v        v         v
+ *                    +--------------------------+
+ *                    |      Transport layer     |
+ *                    +--------------------------+
+ *                                 ^
+ *                               netapi
+ *                                 v
+ *                    +--------------------------+
+ *                    |      Network layer       |
+ *                    +--------------------------+
```

Comment:
(Although - for my definition of 'what this is good for', I'd rather illustrate more (as indicated above) then less.